heuristic/destroy_operators/cross_route.py

```diff
@@ -0,0 +1,64 @@
+from numpy.random import RandomState
+
+from heuristic.classes import Problem, Route, Solution
+from heuristic.functions import customers_to_remove, remove_empty_routes
+
+
+@remove_empty_routes
+def cross_route(current: Solution, rnd_state: RandomState) -> Solution:
+    """
+    Selects two customers that are nearest to each other and their neighbours
+    and removes them from the solution.
+    See ``customers_to_remove`` for the degree of destruction done.
+
+    Similar to cross route removal in Hornstra et al. (2020).
+    """
+    problem = Problem()
+    destroyed = current.copy()
+
+    to_remove = customers_to_remove(problem.num_customers)
+
+    customers = set(range(problem.num_customers))
+    removed = set()
+
+    while len(removed) < to_remove:
+        candidate = rnd_state.choice(tuple(customers))
+
+        route_candidate = destroyed.find_route(candidate)
+        selected_1 = \
+            _select_candidate_and_neighbours(route_candidate, candidate)
+        route_other = []
+        selected_2 = []
```

Comment:
Ugly and does not conform to types. Why not remove candidate and neighbours here, and then remove other later on? That cleans up the flow nicely.